src/cmd/services/m3coordinator/downsample/downsampler.go

```diff
@@ -80,11 +80,13 @@ type SamplesAppenderOverrideRules struct {
 // SamplesAppender is a downsampling samples appender,
 // that can only be called by a single caller at a time.
 type SamplesAppender interface {
-	AppendCounterSample(value int64) error
-	AppendGaugeSample(value float64) error
-	AppendCounterTimedSample(t time.Time, value int64) error
-	AppendGaugeTimedSample(t time.Time, value float64) error
-	AppendTimerTimedSample(t time.Time, value float64) error
+	AppendUntimedCounterSample(value int64) error
+	AppendUntimedGaugeSample(value float64) error
+	AppendUntimedTimerSample(value float64) error
+	AppendCounterSample(t time.Time, value int64) error
+	AppendGaugeSample(t time.Time, value float64) error
+	AppendTimerSample(t time.Time, value float64) error
+	DropTimestamp() bool
```

Comment:
is this only used by tests as well? The multi-appender impl is a bit confusing (in that it returns true, but not all underlying appenders will actually drop the timestamp?

Reply:
Yeah. I could not think of another way to write a test that validates which path metrics end up, other than on the appenders themselves. Let me think a bit more if I have a way to get a targeted test and then I can clean that up.